circuit-foundation/src/commonMain/kotlin/com/slack/circuit/foundation/CircuitContent.kt

```diff
@@ -147,7 +149,9 @@ public fun <UiState : CircuitUiState> CircuitContent(
         onDispose(eventListener::onDisposePresent)
       }
 
-      val state = presenter.present()
+      val pauseablePresenter = remember(presenter) { presenter.toPauseablePresenter() }
```

Comment:
I can't make my mind up whether we want to force people into this or not?

Safest is probably not to, and devs can extend `PauseablePresenter` as they need. This does make testing harder though as we basically need to parameterize all tests so that we cover both bases.

Reply:
Re: opt-in/opt-out - what're the pros and cons of that? Feels like this could safely be default behavior (pausing). We could offer an opt-out at the screen level via marker interface, similar to `@StaticScreen` in #1418, or at the `Circuit` config level. We could also start with it defaulting to off but emphasize in the changelog that we want to make it the default.

Re: `PausablePresenter` at all - what do you think of exposing a composable `pausableState()` function instead of a class? It seems `PausablePresenter` is just a thin wrapper around that sort of logic without requiring a class wrapper.

```kotlin
@Composable
fun <UiState : CircuitUiState> Presenter<UiState>.pauseAwarePresent(
  isPaused: Boolean = LocalLifecycle.current.isPaused
): UiState {
  return pausableState(isPaused) { present() }
}

@Composable
fun <UiState : CircuitUiState> pausableState(
  isPaused: Boolean = LocalLifecycle.current.isPaused,
  produceState: @Composable () -> UiState,
): UiState {
  var uiState by remember(produceState) {
    mutableStateOf<UiState?>(null)
  }

  val saveableStateHolder = rememberSaveableStateHolder(produceState)

  if (!isPaused || uiState == null) {
    // TODO does isPaused need to be captured as a key? Or does it matter if it's a param?
    val retainedStateRegistry = rememberRetained(produceState) { RetainedStateRegistry() }
    CompositionLocalProvider(LocalRetainedStateRegistry provides retainedStateRegistry) {
      saveableStateHolder.SaveableStateProvider("pauseable_presenter") { uiState = produceState() }
    }
  }

  return uiState!!
}
```

Reply:
I like this, refactored.

Reply:
Re: opt-in/out. I also think this is safe, so am comfortable with enabling this as default. I still think we need to add an opt-out though, so added a marker interface